test/gurka/test_filter.cc

```diff
@@ -0,0 +1,108 @@
+#include "baldr/graphreader.h"
+#include "mjolnir/util.h"
+
+#include "gurka.h"
+#include "test/test.h"
+
+#if !defined(VALHALLA_SOURCE_DIR)
+#define VALHALLA_SOURCE_DIR
+#endif
+
+using namespace valhalla;
+using namespace valhalla::baldr;
+using namespace valhalla::gurka;
+
+valhalla::gurka::map the_map = {};
+
+class FilterData : public ::testing::Test {
+protected:
+  valhalla::gurka::map BuildPBF(const std::string& workdir) {
+    constexpr double gridsize_metres = 100;
+
+    const std::string ascii_map = R"(
+		A--------B--------C
+                 |
+                 |
+                 |
+                 D
+    )";
+
+    const gurka::ways ways = {
+        {"ABC",
+         {{"highway", "residential"},
+          {"name:forward", "name in forward dir"},
+          {"name:backward", "name in backward dir"},
+          {"osm_id", "100"}}},
+        {"BD", {{"highway", "footway"}}},
+    };
+
+    if (!filesystem::exists(workdir)) {
+      bool created = filesystem::create_directories(workdir);
+      EXPECT_TRUE(created);
+    }
+
+    constexpr double gridsize = 100;
+
+    const auto layout =
+        gurka::detail::map_to_coordinates(ascii_map, gridsize_metres, {-82.68811, 40.22535});
+
+    auto pbf_filename = workdir + "/map.pbf";
+    detail::build_pbf(layout, ways, {}, {}, pbf_filename);
+
+    valhalla::gurka::map result;
+    result.nodes = layout;
+    return result;
+  }
+};
+
+///////////////////////////////////////////////////////////////////////////////
+TEST_F(FilterData, CheckStreetNamesAfterFilter) {
+
+  const std::string workdir = "test/data/gurka_filter";
+
+  if (!filesystem::exists(workdir)) {
+    bool created = filesystem::create_directories(workdir);
+    EXPECT_TRUE(created);
+  }
+
+  the_map = BuildPBF(workdir);
+
+  const std::string sqlite = {VALHALLA_SOURCE_DIR "test/data/language_admin.sqlite"};
+  the_map.config =
+      test::make_config(workdir,
+                        {{"mjolnir.admin", {VALHALLA_SOURCE_DIR "test/data/language_admin.sqlite"}},
+                         {"mjolnir.tile_dir", workdir + "/tiles"},
+                         {"mjolnir.include_pedestrian", "false"}});
+
+  std::vector<std::string> input_files = {workdir + "/map.pbf"};
+
+  build_tile_set(the_map.config, input_files, mjolnir::BuildStage::kInitialize,
+                 mjolnir::BuildStage::kValidate, false);
```

Comment:
not sure I understand what the test does, can you explain a bit what's going on? e.g. why this elaborate way of building the tiles if we're only skipping the cleanup?

Reply:
again, I just cp'd and pasted from another test.  I will add clean up